modules/gui/include/visp3/gui/vpDisplayFactory.h

```diff
@@ -199,8 +200,100 @@ std::shared_ptr<vpDisplay> createDisplay(vpImage<T> &I, const int winx = -1, con
   return std::shared_ptr<vpDisplay>(nullptr);
 #endif
 }
-#endif
 
+namespace impl
+{
+struct GridSettings
+{
+  unsigned int rows;
+  unsigned int cols;
+  unsigned int startY;
+  unsigned int startX;
+  unsigned int paddingX;
+  unsigned int paddingY;
+};
+
+void makeDisplayGridHelper(std::vector<std::shared_ptr<vpDisplay>> &res, const GridSettings &settings,
+                           unsigned int currRow, unsigned int currCol,
+                           unsigned int currentPixelX, unsigned int currentPixelY,
+                           unsigned int maxRowHeightPixel)
+{
+  if (currRow != (settings.rows - 1)  && (currCol != settings.cols - 1)) {
+    throw vpException(vpException::dimensionError, "Too few images for the grid size");
+  }
+  (void)res;
+  (void)settings;
+  (void)currRow;
+  (void)currCol;
+  (void)currentPixelX;
+  (void)currentPixelY;
+  (void)maxRowHeightPixel;
+}
```

Comment:
This function seams useless or not fully implemented

Reply:
This function is the end condition of the recursion of the function. Since there are no vpImage passed as an argument, we consider that we have iterated over all the displays to create. If the dimensions are incorrect (currCol vs cols and currRow vs rows) then we raise an exception. Otherwise, res is filled with all the created displays, and `makeDisplayGrid` can return the result.